src/main/java/com/fontys/api/service/TournamentService.java

```diff
@@ -80,11 +86,41 @@ public String deleteTournament(Integer id) {
         return Optional.empty();
     }
 
-    private void validateTournament(Integer id) throws InvalidAttributeValueException {
-        if (tournamentRepository.findById(id).isEmpty()) {
+    @Transactional
+    public String addMatchToTournament(Integer tournamentId, Integer matchId)
+    {
+        Optional<Tournament> tournament = tournamentRepository.findById(tournamentId);
+        Optional<Match> match = matchRepository.findById(matchId);
+
+        if(tournament.isEmpty())
+        {
+            return "The tournament does not exist";
+        }
+        if(match.isEmpty())
+        {
+            return "The match does not exist";
+        }
+        Tournament tournament1 = tournament.get();
+        Match match1 = match.get();
+
+        if(tournament1.getMatches().contains(match1))
+        {
+            return "The match is already added to the tournament!";
+        }
+        tournament1.getMatches().add(match1);
+        tournamentRepository.save(tournament1);
+        return "The match was successfully added to the tournament.";
+    }
+
+
+
+    private Tournament validateTournament(Integer id) throws InvalidAttributeValueException {
```

Comment:
Maybe we could stay consistent and have validations return void or an exception.

Reply:
I personally prefer this type of validation over the other. Otherwise you will have warnings that tournament might be empty.

Reply:
Resolving this becuase there in a newer version